team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
So then that'd leave us with

```
if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7')) continue;
```

I'd probably also rename `moveGen` to `fatherLatestMoveGen` or something. In hindsight, renaming `dexEntry` to `father` would make this entire block a lot more readable.